x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/factories/utils/strip_non_ecs_fields.test.ts

```diff
@@ -71,27 +71,6 @@ describe('stripNonEcsFields', () => {
     ]);
   });
 
-  // https://github.com/elastic/sdh-security-team/issues/736
-  describe('fields that exists in the alerts mapping but not in local ECS(ruleRegistry) definition', () => {
```

Comment:
@vitaliidm We are no longer include the `device.*` fields in the ECS component template so this test will fail. I removed it but I could try to replace it with another object field in ECS if you'd prefer.

Reply:
@ymao1 , this test was added specifically to cover SDH issue: https://github.com/elastic/sdh-security-team/issues/736, where customer had problem with `device.*` fields not compatible with ECS.

How changes in PR would affect this use case?

1. Non compatible device field will be saved in alert index, when it was rolled out. This is what we expect, alerts will be always generated.
2. When alerts index is not yet rolled out, but fields non compatible - alerts won't be generated. Since we won't be able to remove non-compatible fields from alert source before saving it.  

If assumption 2 correct, customers might see error covered by test to return, until alerts index rolls out

Reply:
@ymao1 , this test was added specifically to cover SDH issue: https://github.com/elastic/sdh-security-team/issues/736, where customer had problem with `device.*` fields not compatible with ECS.

How changes in PR would affect this use case?

1. Non compatible device field will be saved in alert index, when it was rolled out. This is what we expect, alerts will be always generated.
2. When alerts index is not yet rolled out, but fields non compatible - alerts won't be generated. Since we won't be able to remove non-compatible fields from alert source before saving it.  

If assumption 2 correct, customers might see error covered by test to return, until alerts index rolls out

Reply:
@ymao1 , this test was added specifically to cover SDH issue: https://github.com/elastic/sdh-security-team/issues/736, where customer had problem with `device.*` fields not compatible with ECS.

How changes in PR would affect this use case?

1. Non compatible device field will be saved in alert index, when it was rolled out. This is what we expect, alerts will be always generated.
2. When alerts index is not yet rolled out, but fields non compatible - alerts won't be generated. Since we won't be able to remove non-compatible fields from alert source before saving it.  

If assumption 2 correct, customers might see error covered by test to return, until alerts index rolls out

Reply:
We've changed our approach, so this change is no longer necessary. Thanks for reviewing!